python/cuspatial/cuspatial/core/spatial/distance.py

```diff
@@ -101,14 +101,41 @@ def haversine_distance(p1: GeoSeries, p2: GeoSeries):
     Parameters
     ----------
     p1: GeoSeries
-        Series of points
+        Series of points as floats
     p2: GeoSeries
-        Series of points
+        Series of points as floats
 
     Returns
     -------
     result : cudf.Series
         The distance between pairs of points between `p1` and `p2`
+    >>> import cudf
+    >>> import cuspatial
+    >>> a = {"latitude":[0.0,0.0,1.0,1.0],
+    ...      "longitude": [0.0,1.0,0.0,1.0]}
+    >>> df = cudf.DataFrame(data=a)
+
+    >>> # Create cuSpatial GeoSeries from cuDF Dataframe
+    >>> cuGeoSeries = cuspatial.GeoSeries.from_points_xy(
+    ...     df[['longitude', 'latitude']].interleave_columns()
+    ... )
+
+    >>> # Create Comparator cuSpatial GeoSeries from a comparator point
+    >>> df['compare_lat'] = 2.0
+    >>> df['compare_lng'] = 2.0
+    >>> atlGeoSeries = cuspatial.GeoSeries.from_points_xy(
```

Comment:
Did you mean "alt" as in "alternate" or "atl" as in "Atlanta"? I assume the former. How about we just call this `cmpGeoSeries`?  

Actually we should be using [standard Python naming conventions,](https://peps.python.org/pep-0008/#naming-conventions) so these don't look like class names.

Reply:
agreed, but followed your `cmp_gs` recommendation below